### PR TITLE
fix bug of mixed precision training

### PR DESCRIPTION
**Pull request by denghuilu**
#1469 should be fixed by this PR.

### CODE REVIEW

**Review by wanghan-iapcm — COMMENTED** [review]
Do wen have similar issues in other descriptors?

**Comment by denghuilu** [comment]
> Do wen have similar issues in other descriptors?

No, only `se_a` type support the mixed precision training.

File: deepmd/descriptor/se_a.py [hunk 1]
```diff
@@ -738,7 +738,7 @@ def _filter_lower(
             # we can safely return the final xyz_scatter filled with zero directly
             return tf.cast(tf.fill((natom, 4, outputs_size[-1]), 0.), self.filter_precision)
           # natom x nei_type_i x out_size
-          xyz_scatter = tf.reshape(xyz_scatter, (-1, shape_i[1]//4, outputs_size[-1]))  
+          xyz_scatter = tf.cast(tf.reshape(xyz_scatter, (-1, shape_i[1]//4, outputs_size[-1])), self.filter_precision)
```
Comment (njzjz) [inline deepmd/descriptor/se_a.py:741]: If I understand correctly, it causes the error that [`deepmd.utils.network.embedding_net`](https://github.com/deepmodeling/deepmd-kit/blob/82c787de0ea98e3c032368a5fd4ee46a1fac6283/deepmd/utils/network.py#L99) accepts an input with `self.filter_precision` precision but returns one with `self.mixed_prec`. It is inconsistent. You should both cast and cast back inside the `embedding_net` or outside that method, but do not cast inside the `embedding_net` and cast back here.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: deepmd/utils/network.py [hunk 2]
```diff
@@ -79,15 +79,13 @@ def one_layer(inputs,
...
-                    return tf.reshape(activation_fn(hidden), [-1, outputs_size]) * idt
+                    hidden = tf.reshape(activation_fn(hidden), [-1, outputs_size]) * idt
                 else :
-                    return tf.reshape(activation_fn(hidden), [-1, outputs_size])                    
-        else:
-            if useBN:
-                None
-                # return self._batch_norm(hidden, name=name+'_normalization', reuse=reuse)
-            else:
-                return hidden
```
Comment (wanghan-iapcm) [inline deepmd/utils/network.py:90]: why the case `activation_fn == None` is removed?

**Review by njzjz — APPROVED** [review]

**Pull request merged**